File: hcx-pipeline-jobs/preauth-job/src/main/scala/org/swasth/dp/preauth/task/PreauthConfig.scala
package org.swasth.dp.preauth.task

import com.typesafe.config.Config
import org.swasth.dp.core.job.BaseJobConfig

class PreauthConfig(override val config: Config) extends BaseJobConfig(config, "PreauthJob") {

  private val serialVersionUID = 2905979434303791379L

  val kafkaInputTopic: String = config.getString("kafka.input.topic")

  // Consumers
  val eligibilityCheckConsumer = "preauth-consumer"
  val downstreamOperatorsParallelism: Int = config.getInt("task.downstream.operators.parallelism")

}

File: hcx-pipeline-jobs/preauth-job/src/main/java/org/swasth/dp/preauth/task/PreauthConfig.java
package org.swasth.dp.preauth.task;


import com.typesafe.config.Config;
import org.swasth.dp.core.job.BaseJobConfig;

public class PreauthConfig extends BaseJobConfig {

  private final Config config;

  // kafka
  public String kafkaInputTopic;

  // Consumers
  public String preauthConsumer = "preauth-consumer";
  public int consumerParallelism;
  public int downstreamOperatorsParallelism;

  public PreauthConfig(Config config, String jobName) {
    super(config, jobName);
    this.config = config;
    initValues();
  }

  private void initValues() {
    kafkaInputTopic = config.getString("kafka.input.topic");
    consumerParallelism = config.getInt("task.consumer.parallelism");
    downstreamOperatorsParallelism = config.getInt("task.downstream.operators.parallelism");
  }

}
